### PR TITLE
[RDPEGFX] checks for cache import, prevent 0 size allocations

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: channels/rdpgfx/server/rdpgfx_main.c [hunk 1]
```diff
@@ -1104,9 +1104,9 @@ static UINT rdpgfx_recv_cache_import_offer_pdu(RdpgfxServerContext* context, wSt
 	/* cacheEntriesCount (2 bytes) */
 	Stream_Read_UINT16(s, pdu.cacheEntriesCount);
 
-	if (pdu.cacheEntriesCount <= 0)
+	/* 2.2.2.16 RDPGFX_CACHE_IMPORT_OFFER_PDU */
+	if (pdu.cacheEntriesCount > 5462)
```
Comment (mfleisz) [inline channels/rdpgfx/server/rdpgfx_main.c]: 2.2.2.16 says "MUST be less than 5462" so this check should be `>=`

Comment (akallabeth) [reply channels/rdpgfx/server/rdpgfx_main.c]: good catch

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@hardening 
* `0` size allocations are not valid and will trigger compiler / sanitizer warnings
* The `cacheEntriesCount` value was not checked for validity according to spec. as it is client supplied information in a server side function this should be checked.

**Pull request merged**